lib/APIDigester/ModuleAnalyzerNodes.cpp

```diff
@@ -2509,12 +2509,17 @@ void swift::ide::api::dumpModuleContent(ModuleDecl *MD, StringRef OutputFile,
   opts.Verbose = false;
   SDKContext ctx(opts);
   SwiftDeclCollector collector(ctx);
-  collector.lookupVisibleDecls({MD});
   ConstExtractor extractor(ctx, MD->getASTContext());
-  extractor.extract(MD);
   PayLoad payload;
-  payload.allContsValues = &extractor.getAllConstValues();
-  dumpSDKRoot(collector.getSDKRoot(), payload, OutputFile);
+  SWIFT_DEFER {
+    payload.allContsValues = &extractor.getAllConstValues();
+    dumpSDKRoot(collector.getSDKRoot(), payload, OutputFile);
+  };
+  if (Empty) {
+    return;
+  }
+  collector.lookupVisibleDecls({MD});
+  extractor.extract(MD);
```

Comment:
Works fine obviously, but I think:
```
if (!Empty) {
  collector.lookupVisibleDecls({MD});
  extractor.extract(MD);
  payload.allContsValues = &extractor.getAllConstValues();
}
dumpSDKRoot(collector.getSDKRoot(), payload, OutputFile);
```

Is a little more obvious. `dumpSDKRoot` handles an empty `PayLoad`. Alternatively
```
if (Empty) {
  dumpSDKRoot(collector.getSDKRoot(), OutputFile)
  return;
}
```
Right after the collector is nice too.

Reply:
Thank you for taking a look! @bnbarham The reason I still prefer we set the payload to an empty array of constant values is that the client side, likely written in Swift, can still parse the JSON to get an empty table.